aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group/Program.cs

```diff
@@ -23,33 +35,30 @@
 }
 
 app.MapGet("/", () => "Hello World!");
+app.MapGet("/admin", () => "Authorized Endpoint")
+    .RequireAuthorization("AdminsOnly"); ;
 
-// todo endpoints
-var todos = app.MapGroup("/todos").WithTags("Todo Endpoints").AddRouteHandlerFilter(async (context, next) =>
-{
-    app.Logger.LogInformation("Accessing todo endpoints");
-    return await next(context);
-});
-todos.MapGet("/", RouteHandlers.GetAllTodos);
-todos.MapGet("/{id}", RouteHandlers.GetTodo);
-todos.MapPost("/", RouteHandlers.CreateTodo).AddRouteHandlerFilter(async (context, next) =>
-{
-    // log time taken to process
-    var start = DateTime.Now;
-    var result = await next(context);
-    var end = DateTime.Now;
-    app.Logger.LogInformation($"{context.HttpContext.Request.Path.Value} took {(end - start).TotalMilliseconds}ms");
-    return result;
-});
-todos.MapPut("/{id}", RouteHandlers.UpdateTodo).AddRouteHandlerFilter(async (context, next) =>
-{
-    // log time taken to process
-    var start = DateTime.Now;
-    var result = await next(context);
-    var end = DateTime.Now;
-    app.Logger.LogInformation($"{context.HttpContext.Request.Path.Value} took {(end - start).TotalMilliseconds}ms");
-    return result;
-});
-todos.MapDelete("/{id}", RouteHandlers.DeleteTodo);
+// todoV1 endpoints
+app.MapGroup("/todos/v1")
+   .MapTodosApiV1()
+   .WithTags("Todo Endpoints")
+   .AddEndpointFilter(async (context, next) =>
+   {
+       app.Logger.LogInformation("Accessing todo endpoints");
+       return await next(context);
+   });
+
+// todoV2 endpoints
+app.MapGroup("/todos/v2")
+    .MapTodosApiV2()
+    .WithTags("Todo Endpoints")
+    .AddEndpointFilter(async (context, next) =>
+    {
+        app.Logger.LogInformation("Accessing todo endpoints");
+        return await next(context);
+    });
 
 app.Run();
+
+public partial class Program
```

Comment:
Why do you need this?

Reply:
The `WebApplicationFactory` needs a speakable entry point for the application. The `Program.cs` is less accessible and creates a problem. I found this approach from a StackOverflow question answered by @ardalis

https://stackoverflow.com/questions/69058176/how-to-use-webapplicationfactory-in-net6-without-speakable-entry-point 

Please also refer to this commit by @DamianEdwards 
https://github.com/DamianEdwards/MinimalApiPlayground/commit/2ffc975ee9eddc1587304bad8a88e5620860a392

Reply:
Ah! I see. Thanks.